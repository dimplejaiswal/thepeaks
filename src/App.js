import React, { Suspense } from 'react';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import Navbar from './components/Navbar/Navbar';

const Home = React.lazy(() =>
    import(
        /* webpackPrefetch: true, webpackChunkName: "Home" */ './containers/Home/Home'
    )
);

const Detail = React.lazy(() =>
    import(
        /* webpackPrefetch: true, webpackChunkName: "Detail" */ './containers/Detail/Detail'
    )
);

const Bookmark = React.lazy(() =>
    import(
        /* webpackPrefetch: true, webpackChunkName: "Bookmark" */ './containers/Bookmark/Bookmark'
    )
);

const SearchResult = React.lazy(() => {
    import(
        /* webpackPrefetch: true, webpackChunkName: "SearchResult" */ './containers/SearchResult/SearchResult'
    );
});

const App = () => (
    <BrowserRouter>
        <Navbar>
            <Suspense fallback={<div>Loading...</div>}>
                <Switch>
                    <Route exact path="/" component={Home} />
                    <Route exact path="/detail/:id" component={Detail} />
                    <Route exact path="/bookmark" component={Bookmark} />
                    <Route exact path="/search/:q" component={SearchResult} />
                    <Route render={() => <Redirect push to="/" />}></Route>
                </Switch>
            </Suspense>
        </Navbar>
    </BrowserRouter>
);

export default App;
