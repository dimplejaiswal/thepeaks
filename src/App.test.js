import React from 'react';
import { shallow } from 'enzyme';

// Components
import App from './App';

describe('App Test Suite', () => {
    it('Should render correctly', () => {
        const app = shallow(<App />);
        expect(app).toMatchSnapshot();
    });
});
