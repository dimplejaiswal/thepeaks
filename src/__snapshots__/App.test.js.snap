// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Suite Should render correctly 1`] = `ShallowWrapper {}`;
